backend/api/parse_pdf.py

```
@@ -1066,12 +1134,12 @@ async def process_file_contents_with_semaphore(file_data, name):
         try:
             results = await asyncio.wait_for(
                 asyncio.gather(*[task for _, task in tasks], return_exceptions=True),
-                timeout=300.0  # 5 minute total timeout
+                timeout=720.0  # 12 minute total timeout (allows for parser's 10min + buffer)
             )
         except asyncio.TimeoutError:
             if session_id:
-                progress_manager.error_processing(session_id, "File processing timed out after 5 minutes")
-            raise HTTPException(status_code=504, detail="File processing timed out after 5 minutes")
+                progress_manager.error_processing(session_id, "File processing timed out after 12 minutes")
+            raise HTTPException(status_code=504, detail="File processing timed out after 12 minutes")
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)